src/Native/LdaNative/lda_engine.cpp

```diff
@@ -68,11 +68,11 @@ namespace lda {
         printf("using %d thread(s) to do train/test\n", num_threads_);
 
         bAlphaSumMultiplied = false;
-        atomic_stats_ = new LDAEngineAtomics();
-        model_block_ = new LDAModelBlock();
-        data_block_ = new LDADataBlock(num_threads_);
-        process_barrier_ = new SimpleBarrier(num_threads_);
-        samplerQueue_ = new CBlockedIntQueue();
+        atomic_stats_.reset(new LDAEngineAtomics());
+        model_block_.reset(new LDAModelBlock());
+        data_block_.reset(new LDADataBlock(num_threads_));
+        process_barrier_.reset(new SimpleBarrier(num_threads_));
+        samplerQueue_.reset(new CBlockedIntQueue());
```

Comment:
📝 Using `reset` instead of `std::make_shared` because some of the builds only have C++ 11 (the latter is C++ 14).